cmd/kube-apiserver/app/server.go

```diff
@@ -115,6 +116,9 @@ func Run(s *options.ServerRunOptions) error {
 	if err = s.Authentication.Apply(genericConfig); err != nil {
 		return fmt.Errorf("failed to configure authentication: %s", err)
 	}
+	if err := preflight.RunAPIServerChecks(s); err != nil {
```

Comment:
`utilwait.PollImmediate(retryInterval, retryLimit, etcd.checkEtcdServers)` here?  

`utilwait.PollImmediate(retryInterval, retryLimit, etcd.checkEtcdServers(server list))`.  Since you can return an error from the condition, you could parse inside.  Multiple parses, but for this case it doesn't matter.